src/icq/client.rs

```diff
@@ -82,34 +82,64 @@ pub struct LoginWithPhoneNumberResponseResponse {
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct LoginWithPhoneNumberResponseData {
-    token: LoginWithPhoneNumberResponseToken,
-    host_time: String,
-    session_key: String,
+    pub token: LoginWithPhoneNumberResponseToken,
+    pub host_time: u32,
+    pub session_key: String,
 }
 
 #[derive(Deserialize, Debug)]
 pub struct LoginWithPhoneNumberResponseToken {
-    a: String,
+    pub a: String,
 }
 
 pub async fn send_code(body: &SendCodeBody<'_>) -> Result<SendCodeResponse> {
-    surf::post(SEND_CODE_URL)
-        .with_default_headers()
-        .body_json(&body)
-        .map_err(Error::SerializationError)?
-        .recv_json()
-        .await
-        .map_err(Error::RequestError)
+    json_api(SEND_CODE_URL, body).await
 }
 
 pub async fn login_with_phone_number(
     body: &LoginWithPhoneNumberBody<'_>,
 ) -> Result<LoginWithPhoneNumberResponse> {
-    surf::post(LOGIN_WITH_PHONE_NUMBER_URL)
+    form_api(LOGIN_WITH_PHONE_NUMBER_URL, body).await
+}
+
+async fn form_api<T: serde::Serialize, U: serde::de::DeserializeOwned>(
```

Comment:
The name of this function does not say much. Maybe ``post_form`` would be better, and the other could be named ``post_json``